next/app/coins/[userID]/route.ts

```diff
@@ -0,0 +1,39 @@
+import { headers } from "next/headers";
+import { NextRequest, NextResponse } from "next/server";
+
+const mockLoginDetails: Record<string, { authToken: string, userName: string }> = {
+    alex: {
+      authToken: "123",
+      userName: "alex",
+    },
+    jason: {
+      authToken: "234",
+      userName: "jason",
+    },
+    marie: {
+      authToken: "345",
+      userName: "marie",
+    },
+};
+const mockCoinDetails: Record<string, { balance: number }> = {
+    alex: {
+        balance: 100,
+    },
+    jason: {
+        balance: 200,
+    },
+    marie: {
+        balance: 300,
+    },
+};
+
+export async function GET(_: NextRequest, { params }: { params: { userID: string } }) {
+    // get authorization
+    const headersList = headers();
+    const authToken = headersList.get("authtoken");
+    const authDetails = mockLoginDetails[params.userID];
+    
+    if (authDetails.authToken === authToken) return NextResponse.json({ data: mockCoinDetails[authDetails.userName] });
```

Comment:
For this route, I wanted to show off how middleware could solve this auth issue. I guess you could do it on a per-route basis, but if I remember correctly, you should be able to make a middleware.ts that checks the route before proceeding tp the actual API route.

Pretty much move some of this auth logic to the middleware.ts file.

Reply:
Completed!